test/controllers/businessmatching/updateservice/add/BusinessAppliedForPSRNumberControllerSpec.scala

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.businessmatching.updateservice.add
+
+import controllers.businessmatching.updateservice.UpdateServiceHelper
+import generators.businessmatching.BusinessMatchingGenerator
+import models.businessmatching._
+import models.flowmanagement.{AddServiceFlowModel, BusinessAppliedForPSRNumberPageId}
+import models.status.SubmissionDecisionApproved
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Document
+import org.mockito.Matchers._
+import org.mockito.Mockito._
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.mock.MockitoSugar
+import play.api.i18n.Messages
+import play.api.test.Helpers._
+import services.businessmatching.BusinessMatchingService
+import uk.gov.hmrc.http.cache.client.CacheMap
+import utils.{AuthorisedFixture, DependencyMocks, GenericTestHelper}
+
+import scala.concurrent.Future
+
+class BusinessAppliedForPSRNumberControllerSpec extends GenericTestHelper
+  with MockitoSugar
+  with ScalaFutures
+  with BusinessMatchingGenerator {
+
+  val emptyCache = CacheMap("", Map.empty)
+
+  trait Fixture extends AuthorisedFixture with DependencyMocks {
+    self =>
+
+    val request = addToken(authRequest)
+
+    val mockBusinessMatchingService = mock[BusinessMatchingService]
+    val mockUpdateServiceHelper = mock[UpdateServiceHelper]
+
+    val controller = new BusinessAppliedForPSRNumberController(
+      authConnector = self.authConnector,
+      dataCacheConnector = mockCacheConnector,
+      router = createRouter[AddServiceFlowModel]
+    )
+
+    mockCacheUpdate[AddServiceFlowModel](Some(AddServiceFlowModel.key), AddServiceFlowModel())
+    mockCacheFetch(Some(AddServiceFlowModel(Some(HighValueDealing))))
+    mockApplicationStatus(SubmissionDecisionApproved)
+
+    val businessMatching = businessMatchingGen.sample.get
+
+    when(controller.dataCacheConnector.save[BusinessMatching](any(), any())
```

Comment:
This could use one of the cache helpers: `mockCacheSave[BusinessMatching]`

Reply:
Done